mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_workload_test.go

```diff
@@ -0,0 +1,145 @@
+package apphub_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccDataSourceApphubDiscoveredWorkload_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org_id":          envvar.GetTestOrgFromEnv(t),
+		"random_suffix":   acctest.RandString(t, 10),
+		"billing_account": envvar.GetTestBillingAccountFromEnv(t),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceApphubDiscoveredWorkload_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.google_apphub_discovered_workload.catalog-workload", "name"),
+				),
+			},
+		},
+	})
+}
+
+func testDataSourceApphubDiscoveredWorkload_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_project" "service_project" {
+  project_id ="tf-test-ah-%{random_suffix}"
+  name = "Service Project"
+  org_id = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+
+# Enable Compute API
+resource "google_project_service" "compute_service_project" {
+  project = google_project.service_project.project_id
+  service = "compute.googleapis.com"
+}
+
+resource "time_sleep" "wait_120s" {
+  depends_on = [google_project_service.compute_service_project]
+  create_duration = "120s"
+}
+
+resource "google_apphub_service_project_attachment" "service_project_attachment" {
+  service_project_attachment_id = google_project.service_project.project_id
+  depends_on = [time_sleep.wait_120s]
+}
+    
+data "google_apphub_discovered_workload" "catalog-workload" {
+  location = "us-central1"
+  count=0
```

Comment:
Can you remove this line? This could be the reason for the failed test. It means getting 0 datasource.